packages/gensx/src/checkpoint.ts

```diff
@@ -48,13 +51,27 @@ export interface CheckpointWriter {
 export class CheckpointManager implements CheckpointWriter {
   private nodes = new Map<string, ExecutionNode>();
   private orphanedNodes = new Map<string, Set<ExecutionNode>>();
+  private _secretValues = new Map<string, Set<unknown>>(); // Internal per-node secrets
+  private currentNodeChain: string[] = []; // Track current execution context
+  private readonly MIN_SECRET_LENGTH = 8;
```

Comment:
Why does there need to be a minimum length?

Reply:
Minimums are common in masking implementations, otherwise you might mask an entire object and then end up with a masked value like "a" which has unintended side effects of masking a large fraction of your output. We should definitely keep this, but open to lowering the minimum value if we think 8 chars is too high.